packages/react/hooks.ts

```diff
@@ -243,26 +244,37 @@ export const useDeferredValue = (value: any) => {
 export const useInsertionEffect = useLayoutEffect;
 
 // useSyncExternalStore
-// This implementation is grokked straight from Preact (ref: https://github.com/preactjs/preact/blob/6b92b1fab41599e6da4f96d65b07fdbe0b6ff2fc/compat/src/index.js#L136-L154)
+// This implementation is grokked straight from Preact (ref: https://github.com/preactjs/preact/blob/528a7760c1cdec7515d514c71875860528e6dc12/compat/src/index.js#L142-L171)
 // (c) 2015-present Jason Miller
 export const useSyncExternalStore = (subscribe, getSnapshot) => {
-  const [state, setState] = useState(getSnapshot);
-
   const value = getSnapshot();
 
-  useLayoutEffect(() => {
-    if (value !== state) {
-      setState(() => value);
-    }
-  }, [subscribe, value, getSnapshot]);
+	const [{ instance }, forceUpdate] = useState({
+		instance: { value, getSnapshot }
+	});
```

Comment:
It looks like preact's new `useSyncExternalStore` is now ported from react's `use-sync-external-store/shim`. So you should do it too.

Please do copy their comments. Million.js doesn't have concurrent rendering (Both preact and Million.js performs synchronous rendering just like React pre-18), and it is the only reason why react's shim also works for Million.js (and preact).

https://github.com/facebook/react/blob/main/packages/use-sync-external-store/src/useSyncExternalStoreShimClient.js